### ISSUE TITLE
Run configurable command every X process executions.

### ISSUE THREAD

**Issue by vext01** [softdevteam/krun#150]
TLDR: It would be useful to have the results and log files offloaded to another host every few executions.

We have been relying heavily on partial results for the warmup draft, and ideally we should only ever copy off results while the machine is *not* benchmarking. If we let krun do the copy itself, we can copy at a time which won't affect benchmarks.

The copy could probably be done by scp -- the user would supply (optionally) in the config file: a host string (e.g. `edd@myhost:/path`), a frequency (e.g. every 10 execs), and an SSH key file (as the login would need to be passwordless).

I'm open to other ways of course.

I recommend implementing #143 first to avoid large network copies.

Thoughts?

CC @snim2 @ltratt @cfbolz

**Comment by ltratt** [comment]
I'd suggest generalising this to "run an arbitrary command every *n* execs". The user should do things like set up the ssh keys themselves IMHO, should they want to run `scp`. They might want to do something like "upload via HTTP" (which wouldn't require piddling about with ssh keys).

**Comment by ltratt** [comment]
Related to this, I think the user should be able to get periodic ETA updates (e.g. it would probably be good for us to get ETA updates every couple of days) as Krun's knowledge of the likely ETA improves.

**Comment by vext01** [comment]
> run an arbitrary command every n execs

I like it.

Re: more frequent ETA updates -- this can be handled as part of #125.
